appengine-standard-java8/translate-pubsub/src/main/java/com/example/appengine/translate_pubsub/PubSubPush.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.appengine.translate_pubsub;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(value = "/pubsub/push")
+public class PubSubPush extends HttpServlet {
+  private final JsonParser jsonParser = new JsonParser();
+  private final Gson gson = new Gson();
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    String pubsubVerificationToken = System.getenv("PUBSUB_VERIFICATION_TOKEN");
+    // Do not process message if request token does not match pubsubVerificationToken
+    if (req.getParameter("token").compareTo(pubsubVerificationToken) != 0) {
+      resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+    // parse message object from "message" field in the request body json
+    // decode message data from base64
+    Message message = getMessage(req);
+    try {
+      messageRepository.save(message);
+      // 200, 201, 204, 102 status codes are interpreted as success by the Pub/Sub system
```

Comment:
wouldn't being specific on what a 102 is, be more helpful?

Reply:
Maybe even referencing a constant from somewhere?

Reply:
Added more detail explaining 102 status, but HttpServletResponse doesn't have a constant for 102.